FireboltDotNetSdk.Tests/Integration/SystemEngineTest.cs

```diff
@@ -284,7 +284,91 @@ public void UseEngineTest()
+        {
+            string engineName = "DotNetMixedCaseTest" + suffix;
+            try
+            {
+                CreateCommand("USE ENGINE SYSTEM").ExecuteNonQuery();
+                CreateCommand($"CREATE ENGINE \"{engineName}\"").ExecuteNonQuery();
+                CreateCommand($"USE ENGINE \"{engineName}\"").ExecuteNonQuery();
+                Assert.Throws<FireboltException>(() => CreateCommand($"USE ENGINE {engineName}").ExecuteNonQuery());
+            }
+            finally
+            {
+                executeSafely(
+                    $"USE ENGINE SYSTEM",
+                    $"STOP ENGINE \"{engineName}\"",
+                    $"DROP ENGINE \"{engineName}\""
+                );
+            }
+        }
 
+        [Test]
+        [Category("engine-v2")]
+        [Category("slow")]
+        public void UseEngineMixedCaseToLowerCase()
+        {
+            string engineName = "DotNetToLowerCaseTest" + suffix;
+            try
+            {
+                CreateCommand("USE ENGINE SYSTEM").ExecuteNonQuery();
+                // engine name is lower cased because it is not quoted
+                CreateCommand($"CREATE ENGINE {engineName}").ExecuteNonQuery();
+                CreateCommand($"USE ENGINE {engineName}").ExecuteNonQuery();
+                // engine name remains mixed case and statement fails because engine name was not quoted when we created the engine
+                Assert.Throws<FireboltException>(() => CreateCommand($"USE ENGINE \"{engineName}\"").ExecuteNonQuery());
+            }
+            finally
+            {
+                executeSafely(
+                    $"USE ENGINE SYSTEM",
+                    $"STOP ENGINE {engineName}",
+                    $"DROP ENGINE {engineName}"
+                );
+            }
+        }
+
+        [Test]
+        [Category("engine-v2")]
+        [Category("slow")]
+        public void ConnectToEngineMixedCase()
+        {
+            string engineName = "DotNetMixedCaseTest" + suffix;
+            string databaseName = engineName + "DB";
+            try
+            {
+                CreateCommand("USE ENGINE SYSTEM").ExecuteNonQuery();
+                CreateCommand($"CREATE ENGINE \"{engineName}\"").ExecuteNonQuery();
+                CreateCommand($"CREATE DATABASE \"{databaseName}\"").ExecuteNonQuery();
+
+                // now try to connect to newerly created engine
+                string connectionString = ConnectionString(new Tuple<string, string?>[]
+                {
+                    Tuple.Create<string, string?>(nameof(Engine).ToLower(), engineName),
+                    Tuple.Create<string, string?>(nameof(Database).ToLower(), databaseName)
+                });
+                DbConnection connection = new FireboltConnection(connectionString);
+                connection.Open();
+
+                DbCommand command = connection.CreateCommand();
+                command.CommandText = "SELECT 1";
```

Comment:
I would do CREATE TABLE test (id INT); INSERT INTO test VALUES (1);
This ensures we're 100% connected to the engine/db and not system engine.

Reply:
Done